src/fmm.c

```diff
@@ -1390,7 +1386,7 @@ void fmm_release(void *address)
 			vm_remove_object(&cpuvm_aperture, object);
 		pthread_mutex_unlock(&cpuvm_aperture.fmm_mutex);
 		/* Free the memory from the system */
-		free(address);
+		munmap(address, object->size);
```

Comment:
You're using object after is was freed. You should copy object->size to a local variable before you call vm_remove_object.

Reply:
What about when object is NULL, should we skip the munmap in that case?

Reply:
Good point. Yes, if there is no object, it means the memory wasn't allocated from the Thunk in the first place. So we have no business freeing it.